doc/sphinx/practical-tools/utilities.rst

```diff
@@ -262,6 +268,40 @@ The following makefile rules can be extended.
   One can append lines to the generated ``.merlin`` file extending this
   target.
 
+.. _coqmakefilelocallate:
+
+CoqMakefile.local-late
+++++++++++++++++++++++
+
+The optional file ``CoqMakefile.local-late`` is included at the end of the generated
+file ``CoqMakefile``.  The following is an incomplete list of accessible variables:
+
+:COQ_VERSION:
+   describes the version of ``coqc`` being used, and can be used to
+   provide different behavior depending on what version of Coq is
+   being used
+:COQMAKEFILE_VERSION:
+   describes the version of Coq that was used to generate the
+   Makefile, and can be used to detect version mismatches
+:ALLDFILES:
+   contains the list of generated dependency files, and can be used,
+   for example, to instruct ``make`` to recompute all dependencies
+   when some file changes by writing ``$(ALLDFILES): myfile`` or to
+   indicate that some file must be generated before dependencies can
+   be computed by writing ``$(ALLDFILES): | mygeneratedfile``
+:VOFILES, GLOBFILES, CMOFILES, CMXFILES, OFILES, CMAFILES, CMXAFILES, CMIFILES, CMXSFILES:
+   lists of files that are generated by various invocations of the compilers
+
+Additionally, the following varaibles may be useful for running code
+to decide what targets to present via ``$(shell ...)``; these
+variables are already accessible in the recipe for any rules added in
+``CoqMakefile.local``, but are only accessible to top-level ``$(shell
+...)`` invocations in ``CoqMakefile.local-late``:
```

Comment:
```suggestion
In addition, the following variables may be useful for
to decide what targets to present via ``$(shell ...)``; these
variables are already accessible in the recipe for any rules added in
``CoqMakefile.local``, but are only accessible to top-level ``$(shell
...)`` invocations in ``CoqMakefile.local-late``:
```

"in the recipe"?  Seems imprecise/informal, is there better wording?

Reply:
Perhaps it should be "in the recipes"?  "Recipe" and "rule" are technical terms here, see https://www.gnu.org/software/make/manual/html_node/Recipes.html#Recipes

Reply:
Or "in recipes for rules"

Also "accessible to" -> "accessible from"

Reply:
Done